src/raw_device.rs

```diff
@@ -39,6 +39,48 @@ impl<SpiBus: Bus> RawDevice<SpiBus> {
         Ok(Self { bus, raw_socket })
     }
 
+    /// Enable one or more interrupts
+    ///
+    /// # Args
+    /// * `which` - The interrupts to enable; see `register::socketn::Interrupt`
+    ///             For instance, pass `Interrupt::Receive` to get interrupts
+    ///             on packet reception only.
+    ///
+    pub fn enable_interrupt(&mut self, which: u8) -> Result<(), SpiBus::Error> {
```

Comment:
Would it make more sense to provide an enum of the interrupts that can be enabled instead?

Reply:
Well, the enum exists, it's register::socketn::Interrupt, but (presumably because in Rust you can't do enum|enum and have the result be of type enum), other parts of the code (notably Socket::set_interrupt_mask) take the u8 form instead. Probably the right answer here would be to use the `bitflags` crate, but that's a wider change.

Reply:
Ah yeah indeed, good point thanks. But in an actual driver, you could use `Interrupt::Receive as u8` so the magic number would be ignored, so I think we're fine here as-is.